src/tokenize.cpp

```diff
@@ -1913,6 +1935,21 @@ void tokenize(const deque<int> &data, chunk_t *ref)
             cpd.preproc_ncnl_count++;
          }
 
+         /* Figure out the type of pragma for enable/disable indentation */
+         if (cpd.in_preproc == CT_PP_PRAGMA)
+         {
+            if (memcmp(pc->text(),"asm",3) == 0)
```

Comment:
Why do you need to do this here as well ? Also, this check is not guarded by any new option so it will be called all the times.